packages/di/src/common/decorators/lazyInject.ts

```
@@ -59,16 +67,25 @@ export function LazyInject(
+ * @param key - The named export to import from the module.
+ * @param resolver - A function that returns a promise resolving to the module.
  * @decorator
  */
-export function OptionalLazyInject(key: string, resolver: () => any): PropertyDecorator {
-  return (LazyInject as any)(key, resolver, {optional: true});
+export function OptionalLazyInject(resolver: () => Promise<{default: unknown}>): PropertyDecorator;
+export function OptionalLazyInject(key: string, resolver: () => Promise<{default: unknown}>): PropertyDecorator;
+export function OptionalLazyInject(...args: any[]): PropertyDecorator {
+  const resolver = mapOptions(args, true);
+
+  return (target: any, propertyKey: string | symbol): any | void => {
```

Biome

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)